yarn-project/noir-compiler/src/compile.ts

```
@@ -0,0 +1,105 @@
+import { compile } from '@noir-lang/noir_wasm';
+import nodePath from 'path';
+import fsSync from 'fs';
+import fs from 'fs/promises';
+import noirResolver from '@noir-lang/noir-source-resolver';
+import toml from 'toml';
+import { NoirCompiledContract } from './noir_artifact.js';
+import { ContractAbi, FunctionType } from '@aztec/foundation/abi';
+import { mockVerificationKey } from './mockedKeys.js';
+
+/**
+ * A dependency entry of Nargo.toml.
+ */
+export interface Dependency {
+  /**
+   * Path to the dependency.
+   */
+  path?: string;
+  /**
+   * Git repository of the dependency.
+   */
+  git?: string;
+}
+
+/**
+ * A class that compiles noir contracts and outputs the Aztec ABI.
+ */
+export class ContractCompiler {
+  constructor(private projectPath: string) {}
+
+  /**
+   * Compiles the contracts in projectPath and returns the Aztec ABI.
+   * @returns Aztec ABI of the compiled contracts.
+   */
+  public async compile(): Promise<ContractAbi[]> {
+    const noirContracts = await this.compileNoir();
+    const abis: ContractAbi[] = noirContracts.map(this.convertToAztecABI);
+    return abis;
+  }
+
+  /**
+   * Converts a compiled noir contract to Aztec ABI.
+   * @param contract - A compiled noir contract.
+   * @returns Aztec ABI of the contract.
+   */
+  private convertToAztecABI(contract: NoirCompiledContract): ContractAbi {
+    return {
+      ...contract,
+      functions: contract.functions.map(noirFn => ({
+        name: noirFn.name,
+        functionType: noirFn.function_type.toLowerCase() as FunctionType,
+        parameters: noirFn.abi.parameters,
+        returnTypes: [noirFn.abi.return_type],
+        bytecode: Buffer.from(noirFn.bytecode).toString('hex'),
+        verificationKey: mockVerificationKey,
+      })),
```

Probably overkill for this stage, but it'd be nice to validate the input we're getting from Noir, so we fail loudly here in case the compiler breaks its interface.

---

I thought of using Zod for runtime validation... I'll check it out, the good thing is that if the format of the noir compiler output changes, the snapshot should catch it

---

True, we can just rely on the snapshot for the time being, as long as the user cannot update the underlying noir compiler version we use here.